compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp

```diff
@@ -869,6 +869,48 @@ INTERCEPTOR(void, free, void *ptr) {
   return REAL(free)(ptr);
 }
 
+#if SANITIZER_INTERCEPT_FREE_SIZED
+INTERCEPTOR(void, free_sized, void *ptr, SIZE_T size) {
+  if (DlsymAlloc::PointerIsMine(ptr))
+    return DlsymAlloc::Free(ptr);
+
+  // According to the C and C++ standard, freeing a nullptr is guaranteed to be
+  // a no-op (and thus real-time safe). This can be confirmed for looking at
+  // __libc_free in the glibc source.
+  if (ptr != nullptr)
+    __rtsan_notify_intercepted_call("free_sized");
+
+  if (REAL(free_sized))
```

Comment:
Can you help me understand what the conditions would have to be to be in this interceptor, but not have the REAL `free_sized` available? 

It seems like to be in this interceptor, the real one would have to exist

Reply:
On ELF-based systems, like Linux, the interception code doesn't require that the underlying real function actually exist. This will result in the real function pointer being `nullptr`. In that case, we forward to `free` which we assume exists or we have bigger problems.

Reply:
Cool, thanks